test/unittests/commands/ingest_xmlupload/test_apply_ingest.py

```diff
@@ -0,0 +1,61 @@
+from lxml import etree
+
+from dsp_tools.commands.ingest_xmlupload.apply_ingest import replace_filepath_with_sipi_uuid
+
+# pylint: disable=missing-class-docstring,missing-function-docstring
+
+
+class TestReplaceBitstreamPaths:
+    def test_all_good(self) -> None:
+        xml = (
+            '<knora><resource label="Fluffy Image" restype=":Image2D" id="Fluffy1" '
+            'permissions="res-default"><bitstream permissions="prop-default">images/Fluffy.jpg</bitstream>'
+            "</resource></knora>"
+        )
+        root = etree.fromstring(xml)
+        reference_dict = {"images/Fluffy.jpg": "fluffy_uuid"}
+        res_tree, res_msg = replace_filepath_with_sipi_uuid(root, reference_dict)  # type: ignore[arg-type]
+        assert not res_msg.media_no_uuid
+        assert not res_msg.unused_media_paths
+        expected = (
+            b'<knora><resource label="Fluffy Image" restype=":Image2D" id="Fluffy1" '
+            b'permissions="res-default"><bitstream permissions="prop-default">fluffy_uuid</bitstream>'
+            b"</resource></knora>"
+        )
+        assert etree.tostring(res_tree) == expected
```

Comment:
That is really weird because for me it often does not work, sometimes it does for whatever reasons.

Reply:
It is indeed tricky to not getting knocked out by additional spaces and/or newlines. It took me a while to figure out the suggestion I gave you in my first comment :-)

But I think it's worth it, because unformatted XML-strings are horror :-)

Reply:
No what I actually mean is that tests that have not failed in the past, now suddenly start to fail...

Reply:
The main danger of failing tests when using multiline strings (at least for me) has always been the fact that there may be different whitespace (newlines that are easy to forget, indentation spaces, etc.), so the best thing I found was not to compare strings in tests (whitespace shouldn't matter!) but to compare actual information content wherever possible.  
In this case, you actually compare two XML strings, one of which you let lxml serialize and the other one is hard coded. Why not ask lxml for the thing you're actiually interested in (namely the text value inside the `<bitstream>` tag) which will either be `"images/Fluffy.jpg"` or `"fluffy_uuid"`, depending on if the function does what it should - so you could just assert that.

Reply:
(maybe not the most clear comment from my side... sorry! does it make sense?)

Reply:
It does to me, thanks! The only thing that could be a problem with your method is that we don't check, that the rest of the string is unchanged...

Reply:
that's true - I'm not sure we have to test that, though, because unless there is good reason why we think it might (in which case that should be tested specifically), it's essentially just a guessing/hoping game, if the test string fulfils the condition where this might happen. But it's up to you if you deem that necessary